File: ServiceExampleJava/app/src/main/java/com/rubenmimoun/serviceexamplejava/MainActivity.java
package com.rubenmimoun.serviceexamplejava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity  {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button start  =  findViewById(R.id.start) ;
        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService();
            }
        });
        Button stop =  findViewById(R.id.stop);
        stop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                stopService();
            }
        });
    }



    private void startService(){
        Intent intent = new Intent(this, MyService.class);
        startService(intent);
    }


    private void stopService(){
        Intent intent =  new Intent(this, MyService.class);
        stopService(intent);
    }


}

File: ServiceExampleKotlin/app/src/main/java/com/rubenmimoun/serviceexamplekotlin/MainActivity.kt
package com.rubenmimoun.serviceexamplekotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        start.setOnClickListener { startService() }
        stop.setOnClickListener { stopService() }

    }


    private fun startService(){
        startService(Intent(this,MyService::class.java))
    }

    private fun stopService(){
        stopService(Intent(this, MyService::class.java))
    }
}

